src/kernel/caches.h

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2024-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_KERNEL_CACHES_H
+#define BITCOIN_KERNEL_CACHES_H
+
+#include <util/check.h>
+
+#include <algorithm>
+#include <bit>
+#include <cstdint>
+#include <limits>
+
+//! Suggested default amount of cache reserved for the kernel (MiB)
+static constexpr int64_t DEFAULT_KERNEL_CACHE{450};
+//! Max memory allocated to block tree DB specific cache (MiB)
+static constexpr int64_t MAX_BLOCK_DB_CACHE{2};
+//! Max memory allocated to coin DB specific cache (MiB)
+static constexpr int64_t MAX_COINS_DB_CACHE{8};
+
+//! Guard against truncation of values before converting.
+constexpr size_t MiBToBytes(int64_t mib)
+{
+    Assert(std::countl_zero(static_cast<uint64_t>(mib)) >= 21); // Ensure signed bit is unset + enough zeros to shift.
```

Comment:
nit: If you re-touch, slightly more correct:
```suggestion
    Assert(std::countl_zero(static_cast<uint64_t>(mib)) >= 21); // Ensure sign bit is unset + enough zeros to shift.
```

Reply:
Should probably fix this typo in commit it was introduced (a5742b7c09f1b4204abf4a1992018fb7be2fef27), instead of in final commit (224bfeb07f7561470451c95dc0df1db959bc187e).